File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/ExplanationDocumentAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class ExplanationDocumentAddRequest {

    @NotNull
    private final Date dateOfFill;

    @NotNull
    private final Time timeOfFill;

    @NotNull
    @NotEmpty
    private final String placeOfFill;

    @NotNull
    private final Long policeOfficerID;

    private final Long carAccidentParticipant;

    private final Long carAccidentWitness;

    @NotNull
    @NotEmpty
    private final String interviewedPersonType;

    @NotNull
    private final Long carAccidentEntityID;

}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/ExplanationDocumentAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class ExplanationDocumentAddRequest(
    val dateOfFill: Date,
    val timeOfFill: Time,
    val placeOfFill: String,
    val policeOfficerID: Long,
    val carAccidentParticipant: Long?,
    val carAccidentWitness: Long?,
    val interviewedPersonType: String,
    val carAccidentEntityID: Long
)
